plugin-management-library/src/main/java/io/jenkins/tools/pluginmanager/impl/PluginManager.java

```diff
@@ -1056,15 +1060,36 @@ public String getAttributeFromManifest(File file, String key) {
         return null;
     }
 
+    /**
+     * Gets Jenkins version using one of the available methods.
+     * @return Jenkins version or {@code null} if it cannot be determined
+     */
+    @CheckForNull
+    public VersionNumber getJenkinsVersion() {
+        if (jenkinsVersion != null) {
+            return jenkinsVersion;
+        }
+        if (jenkinsWarFile != null) {
```

Comment:
what do you think about moving this to the CLI? so the lib only works with a resolved jenkins version?

Reply:
I do not think that Jenkins version is always needed for the library's logic. OTOH we could be more opinionated there